test/e2e/service.go

```diff
@@ -230,36 +235,34 @@ var _ = Describe("Services", func() {
 				Expect(err).NotTo(HaveOccurred())
 			}
 		}()
+		const timeout = 4 * time.Minute
 
-		name1 := "test1"
-		addEndpointPodOrFail(c, ns, name1, labels)
-		names = append(names, name1)
+		for start := time.Now(); time.Since(start) < timeout; {
```

Comment:
Please use our standard util.Poll() retry mechanism here, rather than a homegrown loop.
See PR #7858 for an example of how to use util.Poll().